src/sdcard.c

```diff
@@ -4,9 +4,11 @@
 #include <stdio.h>
 #include <string.h>
 
+#define SDCARD_ERROR 0xff;
+
 uint8_t sector_buffer[512];
 
-const long sd_sectorbuffer = 0xffd6e00L;
+const uint32_t sd_sectorbuffer = 0xffd6e00U;
```

Comment:
This probably works, but I wonder why `U` is used, to silence a warning? Maybe try without the `U` or use `UL` as technically it is a 16 bit unsigned constant (without the `L`).

Reply:
Here the main change is from signed to unsigned 32 bit, so it should be UL. Will fix that. In any case there's no loss in precision so no complaints from the compiler.